File: Invert binary Tree/Python/solution1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if root is None:
            return root
        # get the right side swapped
        left = self.invertTree(root.right)
        # get the left side swapped and updated to right
        root.right = self.invertTree(root.left)
        # swap the right to be left
        root.left = left
        
        return root

File: Invert binary Tree/C#/solution1.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public TreeNode InvertTree(TreeNode root) {
        if(root != null){
            helper(root);
        }
        return root;
    }
    
    private void helper(TreeNode root) {
        if(root == null){
            return;
        }
        // recurse until we get to a leaf node
        helper(root.left);
        helper(root.right);
        // Swap the left and right nodes
        TreeNode temp = root.left;
        root.left = root.right;
        root.right = temp;
    }
}